source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1416,27 +1429,25 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
     // Processing code uses this specific error code in the case that a
     // message was received out of order.
     stats_.spurious_msgs_received_.inc();
-    // When a message is received out of order, ignore it and also
-    // ignore the stream for the rest of this filter instance's lifetime
-    // to protect us from a malformed server.
     ENVOY_STREAM_LOG(warn, "Spurious response message {} received on gRPC stream",
                      *decoder_callbacks_, static_cast<int>(response->response_case()));
-    closeStream();
-    clearAsyncState();
-    processing_complete_ = true;
+    if (config_->failureModeAllow()) {
+      // When a message is received out of order,and fail open is configured,
```

Comment:
This changes the semantic of failure_mode_allow, as it now includes ext_proc server response path.

Should we introduce a new config proto config, rather than overloading the existing field , which could break people's mental model of this field as well).

Reply:
I thought about the new flag approach before.

Currently, the failure_mode_allow control the fail_open/fail_close behavior in gRPC error case and message timeout case.  Introducing a new flag to let it control server mis-behavior case probably brings code and configuration complexity. So, I hope we can keep it simple to have one flag control fail-open/fail-close for the entire external processing logic instead of adding new flags for different error cases.

Reply:
Yeah, I understand the benefits of  one flag but this technically is a breaking change to existing customers?

I don't have super strong opinion on whether one flag or two flags, I just mainly just want to call out the behavior breakage.

I would like to hear from other code owners, if they agree with one flag, please at least update the protocol comments

Reply:
@tyxia this is a good point. Since this change can be user visible we need to add a runtime flag for it.